flycheck-haskell.el

```diff
@@ -257,7 +257,11 @@ buffer."
                 (append .extensions .languages
                         flycheck-ghc-language-extensions))
     (setq-local flycheck-ghc-args
-                (append .other-options flycheck-ghc-args))))
+                (append .other-options
+                        (cons "-hide-all-packages"
+                              (mapcar (apply-partially #'concat "-package ")
+                                      .dependencies))
```

Comment:
I've lived with this for almost one month, I'm using my modified version exclusively. The only noticeable difference is that you need to install all packages including those for test suites before you start working on something, otherwise you'll get a message about missing packages. For me, it's OK.